src/Features/CSharp/Portable/Completion/KeywordRecommenders/AbstractSpecialTypePreselectingKeywordRecommender.cs

```diff
@@ -0,0 +1,28 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CSharp.Extensions.ContextQuery;
+
+namespace Microsoft.CodeAnalysis.CSharp.Completion.KeywordRecommenders
+{
+    internal abstract class AbstractSpecialTypePreselectingKeywordRecommender : AbstractSyntacticSingleKeywordRecommender
+    {
+        public AbstractSpecialTypePreselectingKeywordRecommender(
+            SyntaxKind keywordKind, 
+            bool isValidInPreprocessorContext = false, 
```

Comment:
Tried this out, will require tons of fixups which I don't want to include in this PR:

https://github.com/dotnet/roslyn/issues/12112